src/common.h

```diff
@@ -27,6 +29,53 @@
 
 #define MASK(n) (~((~0U << (n))))
 
+#if defined(_MSC_VER)
+#include <intrin.h>
+static inline int rv_clz(uint32_t v)
+{
+    uint32_t leading_zero = 0;
+    if (_BitScanReverse(&leading_zero, v))
```

Comment:
Since the caller ensures the parameter passed to `rv_clz()` is non-zero before calling it, we don't need to check the return value of `_BitScanReverse()`.

Reply:
> Since the caller ensures the parameter passed to `rv_clz()` is non-zero before calling it, we don't need to check the return value of `_BitScanReverse()`.

I would prefer not to change it, as this return value checking could potentially serve as a debugging method for correctly calling `rv_clz` in future development.

Reply:
> > Since the caller ensures the parameter passed to `rv_clz()` is non-zero before calling it, we don't need to check the return value of `_BitScanReverse()`.
> 
> I would prefer not to change it, as this return value checking could potentially serve as a debugging method for correctly calling `rv_clz` in future development.

For debugging purposes, I prefer adding `assert(v);` and removing the `_BitScanReverse()` check. This way, errors with `v == 0` can be caught. When performance matters, we can compile with `-DNDEBUG` to ignore the check. Currently, when using gcc or clang to compile, the same checks are not performed. If we want to enforce this check, it should be consistent across gcc, clang, and the fallback implementations.

Reply:
> For debugging purposes, I prefer adding assert(v); and removing the _BitScanReverse() check. This way, errors with v == 0 can be caught.

This is better.